diff --git a/bin/rootwrap.c b/bin/rootwrap.c
index 03598a9..450e928 100644
--- a/bin/rootwrap.c
+++ b/bin/rootwrap.c
@@ -229,6 +229,13 @@ int main (int argc, char *argv[])
     setrlimit (RLIMIT_RTPRIO, &rlim);
 #endif
 
+#ifdef BCM2835_INIT_WRAPPER
+    if (!bcm2835_init()) {
+        fprintf (stderr, "Cannot initialize bcm2835 library!\n");
+        exit (1);
+    }
+#endif
+
     uid_t uid = getuid ();
     if (uid == 0)
     {
diff --git a/configure.ac b/configure.ac
index a273d18..fb9ef37 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1480,6 +1480,20 @@ AS_IF([test "${enable_run_as_root}" = "yes"],[
 ])
 
 dnl
+dnl Initialize bcm2835 library in rootwrapper.c
+dnl
+AC_ARG_ENABLE(bcm2835-init,
+  [AS_HELP_STRING([--enable-bcm2835-init],
+    [initialize bcm2835 library in vlc-wrapper (default disabled)])])
+AS_IF([test "${enable_bcm2835_init}" = "yes"],[
+    AC_CHECK_LIB(bcm2835, bcm2835_init, [
+      VLC_ADD_LIBS([vlc_wrapper],[-lbcm2835])
+      AC_DEFINE(BCM2835_INIT_WRAPPER, 1,
+                [Define to 1 to initalize bcm2835 library in root wrapper.])],
+      [ AC_MSG_ERROR([Cannot find bcm2835 library...]) ])
+])
+
+dnl
 dnl Stream output
 dnl
 AC_ARG_ENABLE(sout,
